File: src/test/java/io/tripled/social/client/application/SpiedFollowUserUseCase.java
package io.tripled.social.client.application;

public class SpiedFollowUserUseCase implements FollowUserUseCase {

  public String userName;
  public String userNameToFollow;

  @Override
  public String follow(String userName, String userNameToFollow) {
    this.userName = userName;
    this.userNameToFollow = userNameToFollow;

    return userName + " will follow " + userNameToFollow;
  }
}

File: src/test/kotlin/io/tripled/social/client/application/SpiedFollowUserUseCase.kt
package io.tripled.social.client.application

class SpiedFollowUserUseCase : FollowUserUseCase {
    @JvmField
    var userName: String? = null
    @JvmField
    var userNameToFollow: String? = null

    override fun follow(userName: String, userNameToFollow: String): String {
        this.userName = userName
        this.userNameToFollow = userNameToFollow

        return userName + " will follow " + userNameToFollow
    }
}
